deployment/terraform/engine.go

```diff
@@ -89,16 +91,30 @@ func (t *Terraform) runCommand(dst io.Writer, args ...string) error {
 }
 
 func checkTerraformVersion() error {
-	if out, err := exec.Command("terraform", "version").Output(); err != nil {
+	out, err := exec.Command("terraform", "version").Output()
+	if err != nil {
 		return fmt.Errorf("could not run %q command: %w", "terraform version", err)
-	} else if re := regexp.MustCompile(`v\d+.?\d+`); !re.Match(out) {
+	}
+
+	re := regexp.MustCompile(`v\d+.?\d+`)
+	if !re.Match(out) {
 		return fmt.Errorf("could not parse terraform command output: %s", out)
-	} else if version, err := strconv.ParseFloat(string(re.FindAll(out, 1)[0][1:]), 64); err != nil {
+	}
+
+	version, err := strconv.ParseFloat(string(re.FindAll(out, 1)[0][1:]), 64)
+	if err != nil {
 		return fmt.Errorf("could not parse terraform command output: %w", err)
-	} else if version < 0.12 {
-		return fmt.Errorf("at least terraform v0.12 is required, you have %.2f", version)
-	} else if version > 0.12 {
-		mlog.Warn(fmt.Sprintf("This tool tested with terraform v0.12, you have v%.2f. Proceed with your own risk.", version))
+	}
+	if version < minSupportedVersion {
+		return fmt.Errorf("minimum version of terraform %.2f is required, you have %.2f", minSupportedVersion, version)
+	} else if version > maxSupportedVersion {
```

Comment:
I thought we decided not to use else ifs anymore. Since we return, we can split this into two more readable if blocks.

Reply:
I thought what we settled on was that one `if-else-if` or one `if-else` block is okay. But more than one `else-if` is too much. 3 or more if/else-ifs usually point to a switch-case as we agreed on earlier. :P

So IMO it looks like:
- One single if `err != nil` block or an additional else/else-if block with it is fine.
- If there are more than one else-if blocks coming with it, split it.
- If there are more than one else-if in general, move to a switch case.

But I'm 0/5 on whatever as long as we are consistent. Just 1/5 on the 2nd point.

Reply:
That sounds fair. But here my reasoning is that we return early in all cases so there's no real need for the else. That's why I'd prefer splitting it and keep consistency with the rest of the function.

Reply:
No worries.